hc-added contributing.md to repo (#21)
kjaymiller/github-community-standards: CONTRIBUTING.md

# Contributing

When contributing to this repository, please first discuss the change you wish to make via **issue**, email, or any other method with the owners of this repository before making a change.

Please note and visit our [Code of Conduct here!](https://github.com/tnfdesigns/themeanager/blob/d3213d4c7a2bbe7dec607c8562b768e1c540f061/CODE_OF_CONDUCT.md "See Our Code of Conduct"). Be sure to follow it in all your interactions with the project.

> ## Pull Request Process:

Reviewer: I would put the Issues section(s) ahead of the Pull Request section

1. Ensure any install or build dependencies are removed before the end of the layer when doing a build.
2. Update the README.md with details of changes to the interface, this includes new environment variables, exposed ports, useful file locations and container parameters.
3. Increase the version numbers in any examples files and the README.md to the new version that this Pull Request would represent. The versioning scheme we use is [SemVer](http://semver.org/).
4. You may merge the Pull Request in once you have the sign-off of two other developers, or if you do not have permission to do that, you may request the second reviewer to merge it for you.

## New Issue Creation Process:

1. Ensure that the issue is not already created.
2. Create a new issue with the appropriate template.

> 3. Verify that the issue is not already assigned to someone else.

Reviewer: This seems like something that should be in claiming an existing issue. I would create a new section that mentions that

```suggestion

## Claiming an Issue
1. Verify that the issue is not already assigned to someone else.
```

Author: added this section and expanded your traing of thought, I believe.

> 4. If you are a contributor, you may assign yourself to the issue.

Reviewer: ```suggestion
4. If you are a Collaborator, you may assign yourself to the issue.
```

> 4. If you are a contributor, you may assign yourself to the issue.

Reviewer: ```suggestion
4. If you are a Collaborator, you may assign yourself to the issue.
```

Author: got it.

> 4. If you are a contributor, you may assign yourself to the issue.

Reviewer: ```suggestion
2. If you are a contributor, you may assign yourself to the issue.
```

Author: understood i think!

> 5. If you are not a contributor, you may request to be assigned to the issue.

Reviewer: ```suggestion
5. If you are not a Collaborator, you may request to be assigned to the issue.
```

> 5. If you are not a contributor, you may request to be assigned to the issue.

Reviewer: ```suggestion
3. If you are not a Collaborator, you may request to be assigned to the issue.
```

Author: believe this is fixed.

> 6. You may begin working on the issue once you have been assigned to it.

Reviewer: ```suggestion
4. You may begin working on the issue once you have been assigned to it.
```

Author: got it.

> 7. Be sure to attach all issues to the appropriate pull request.

Reviewer: ```suggestion
5. Be sure to attach all issues to the appropriate pull request.
```

# Contributing

First, we would like to thank you for taking the time to contribute to this project. We really appreciate it. We hope to make this a pleasant and fun experiece for you!

When contributing to this repository, please first discuss the change you wish to make via the **issue comments** you are interested in working on, before making any changes.

Please note and visit our [Code of Conduct here!](https://github.com/Github-Community-Standards/github-community-standards/blob/main/CODE_OF_CONDUCT.md "Github-Community-Standards Code of Conduct"). Thank you in advance for being sure to follow it in all your interactions with the project.

## Claiming/Assinging an Issue:

1. First, **_always verify_** that the issue is not already assigned to anyone.

> 1. If you are a [Collaborator](https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-personal-account-on-github/managing-access-to-your-personal-repositories/inviting-collaborators-to-a-personal-repository "Github Collaborator Docs"), you may assign the issue to yourself.

Reviewer: ```suggestion
2. If you are a [Collaborator](https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-personal-account-on-github/managing-access-to-your-personal-repositories/inviting-collaborators-to-a-personal-repository "Github Collaborator Docs"), you may assign the issue to yourself.
```

> 1. If you are a [Contributor](https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/setting-guidelines-for-repository-contributors "Github Contributor Docs"), you may request the issue be assigned to you.

Reviewer: ```suggestion
3. If you are a [Contributor](https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/setting-guidelines-for-repository-contributors "Github Contributor Docs"), you may request the issue be assigned to you.
```

> 1. You may begin working on the issue once the issue has been assigned to you.

Reviewer: ```suggestion
4. You may begin working on the issue once the issue has been assigned to you.
```

> 1. Be sure to attach **_all related issues_** to the appropriate pull request using [Github linking](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue "Github linking ").

Reviewer: ```suggestion
5. Be sure to attach **_all related issues_** to the appropriate pull request using [Github linking](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue "Github linking ").
```

## New Issue Creation Process:

1. First, **_always verify_** the issue has **_not_** already been created in the repository.
   - Failing to verify issues are only created once, almost always leads to duplicate issues and an unorganized repository.

> 1. Create new issues using the appropriate [ISSUE_TEMPLATE](https://docs.github.com/en/issues/tracking-your-work-with-issues/creating-an-issue#creating-an-issue-from-a-repository "Github Issue_Template docs").

Reviewer: ```suggestion
2. Create new issues using the appropriate [ISSUE_TEMPLATE](https://docs.github.com/en/issues/tracking-your-work-with-issues/creating-an-issue#creating-an-issue-from-a-repository "Github Issue_Template docs").
```

Author: My bad, should've maybe asked. I don't remember where I picked it up, but if you number all your items in the numbered list with 1., no matter how many you add it will adjust to the correct number in the list. However it does not show up in the markdown. When looking at the md in preview mode, it does show the correct numbers. I'll go ahead and get them all changed back.

   - All contributors using the provided issue templates, helps us all to understand each issue better.
   - If you are unsure which template to use, please use the **_blank_** template, paying extra attention to being detailed and thorough in your issue description.

## Pull Request Process:

1. Ensure any install or build dependencies are removed before the end of the layer when doing a build.

> 1. Update the README.md with details of changes to the interface, this includes new environment variables, exposed ports, useful file locations and container parameters.

Reviewer: ```suggestion
2. Update the README.md with details of changes to the interface, this includes new environment variables, exposed ports, useful file locations and container parameters.
```

> 1. Increase the version numbers in any examples files and the README.md to the new version that this Pull Request would represent. The versioning scheme we use is [SemVer](http://semver.org/ "Semver.org").

Reviewer: ```suggestion
3. Increase the version numbers in any examples files and the README.md to the new version that this Pull Request would represent. The versioning scheme we use is [SemVer](http://semver.org/ "Semver.org").
```

> 1. You may merge the Pull Request in once you have the sign-off of two other developers, or if you do not have access, you may request the second reviewer to merge it for you.

Reviewer: ```suggestion
4. You may merge the Pull Request in once you have the sign-off of two other developers, or if you do not have access, you may request the second reviewer to merge it for you.
```
